src/pyscipopt/scip.pxi

```diff
@@ -975,6 +975,130 @@ cdef class Constraint:
         return (self.__class__ == other.__class__
                 and self.scip_cons == (<Constraint>other).scip_cons)
 
+cdef class Decomposition:
+    """Base class holding a pointer to SCIP decomposition"""
+    def __init__(self):
+        self.create()
+    @staticmethod
+    cdef create(SCIP_DECOMP* scip_decomp):
+        if scip_decomp == NULL:
+            raise Warning("cannot create Constraint with SCIP_CONS* == NULL")
+        decomp = Decomposition()
+        decomp.scip_decomp = scip_decomp
+        return decomp
+
+    def isOriginal(self):
+        return SCIPdecompIsOriginal(self.scip_decomp)
+
+    def getAreaScore(self):
```

Comment:
As this isn't standard knowledge (or at least I don't know it), if you can add a one line comment for what the function does then please do. This really helps out users and means they don't need to go through SCIP itself for answers.

Reply:
This extends to most functions below